ocrd-import

```diff
@@ -165,104 +166,107 @@ for file in $(find -L . -type f -not -name mets.xml -not -name "*.log" | sort);
-            group=OCR-D-SEG-ALTO
-            if ! ((numpageid)); then
-                base=${base/OCR-D-IMG/$group}
-            fi
-        elif (($ignore)); then
-            warning "unknown type of file '$file'"
-            exit #continue
-        else
-            critical "unknown type of file '$file'"
-            false
-        fi
-        ;;
-        application/pdf|application/postscript|application/oxps|image/x-*|"")
-        case "$suffix" in
-            .pdf|.PDF)
-                inopts=(-units PixelsPerInch -density $((2*$dpi)))
-                outopts=(-background white -alpha remove -alpha off -colorspace Gray -units PixelsPerInch -resample $dpi -density $dpi)
-                ;;
-            *)
-                inopts=()
-                outopts=()
-        esac
-        if (($convert)) && \
-               mkdir -p OCR-D-IMG && \
-               warning "converting '$file' to 'OCR-D-IMG/${base}_*.tif' prior to import" && \
-               convert "${inopts[@]}" "$file" "${outopts[@]}" OCR-D-IMG/"${base}_%04d.tif"; then
-            mimetype=image/tiff
-            IFS=$'\n'
-            files=($(find OCR-D-IMG -name "${base}_[0-9]*.tif" | sort))
-            IFS=$' \t\n'
-            info "converted '$file' to 'OCR-D-IMG/${base}_*.tif' (${#files[*]} files)"
-            if ((${#files[*]}>1)); then
-                for file in "${files[@]}"; do
-                    file="${file#./}"
-                    base="${file%.tif}"
-                    base="${base#OCR-D-IMG/}"
-                    add_file $group $mimetype ${page}_${base:(-4)} "$base" "$file"
-                done
-                # there's no danger of clashes with other files here
-                exit # continue
-            else
-                file="${files[0]}"
-                file="${file#./}"
-            fi
-        elif (($ignore)); then
-            warning "unknown type of file '$file'"
-            exit # continue
-        else
-            critical "unknown type of file '$file'"
-            false
-        fi
-        ;;
-    esac
-    IFS=$'\n'
-    clashes=($(ocrd workspace find -i "$base" -k local_filename -k mimetype -k pageId))
-    IFS=$' \t\n'
-    n=0
-    for clash in "${clashes[@]}"; do
-        let n++ || true
-        IFS=$'\t'
-        fields=($clash)
-        IFS=$' \t\n'
-        # if image, allow PAGE with matching basename
-        # if PAGE, allow image with matching basename
-        if if test $group = OCR-D-IMG; then
-               test "x${fields[1]}" = x${MIMETYPE_PAGE}
-           else [[ "${fields[1]}" =~ image/ ]]
-           fi; then
-            # use existing pageId
-            page=${fields[2]}
-            # use new file ID
-            base="$(basename "$file")" # (including suffix)
-            base="${base// /_}"
-        else
-            warning "files '$file' ($mimetype) and '${fields[0]}' (${fields[1]}) have the same basename"
-        fi
-    done
+    #case "$mimetype" in
+    #    ${MIMETYPE_PAGE})
```

Comment:
If you want to have a version with next to no checks, the right approach would be to just add a switch to circumvent the no-clash check for resulting pages/files. Everything else is already fast (like the PAGE vs ALTO check for .xml) or can be deactivated (like the `--no-convert` switch).

Reply:
Yes, this was very broad commenting out. The "offending" call that slows down import is 

```bash
clashes=($(ocrd workspace find -i "$base" -k local_filename -k mimetype -k pageId))
```